core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -161,42 +184,22 @@
       // invalid scheme. Just fall through and will try "dns:///foo.googleapis.com:8080"
       uriSyntaxErrors.append(e.getMessage());
     }
-    if (nameResolver == null) {
+    // It may be Opaque if the host was parsed as the scheme
+    if (targetUri == null || targetUri.isOpaque()) {
```

Comment:
From what I can tell, checking host is equivalent to checking `isOpaque()`. Given the definition of target, I see no way to avoid multiple calls to `newNameResolver`, and that was understood when expanding it from "simple string" to a URI-scheme based system.